### PR TITLE
Add keyboard firmware treadstone32lite

### PR DESCRIPTION
**Pull request by marksard**
I want add keyboard firmware for Treadstone32 lite

## Description

I make a variation the Treadstone32 PCB. It keyboard is Treadstone32 lite, it use microcontroller is a pro micro. And change a pin assign. Therefore it need for update firmware.
And  I update some keymaps.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/treadstone32/rev1/rev1.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	// runs for every action, just before processing by the firmware
+
+	return process_record_user(keycode, record);
+}
+
+void led_set_kb(uint8_t usb_led) {
+	// put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+
+	led_set_user(usb_led);
+}
```
Comment (drashna) [inline keyboards/treadstone32/rev1/rev1.c:43]: These aren't actually doing anything, and can be removed. 

```suggestion
```

File: keyboards/treadstone32/readme.md [hunk 2]
```diff
@@ -11,8 +11,11 @@ Hardware Availability: links to where you can find this hardware
 Make example for this keyboard (after setting up your build environment):
 
     make treadstone32:default:dfu
+    or
+    make treadstone32/lite:default:avrdude
```
Comment (fauxpark) [inline keyboards/treadstone32/readme.md:15]:
```suggestion
    make treadstone32:default
    or
    make treadstone32/lite:default
```

File: keyboards/treadstone32/rules.mk [hunk 3]
```diff
@@ -27,3 +27,6 @@ UNICODE_ENABLE = no         # Unicode
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
 RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight.
 LEADER_ENABLE = no
+
```
Comment (fauxpark) [inline keyboards/treadstone32/rules.mk:30]:
```suggestion
```

File: keyboards/treadstone32/lite/config.h [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+  #undef MOUSEKEY_MAX_SPEED
+  #define MOUSEKEY_MAX_SPEED 3
+
+  #undef MOUSEKEY_MOVE_DELTA
+  #define MOUSEKEY_MOVE_DELTA 4
+
+  #undef MOUSEKEY_DELAY
+  #define MOUSEKEY_DELAY 0
+#endif
```
Comment (zvecr) [inline keyboards/treadstone32/lite/config.h:112]: Should really use defaults where possible at the keyboard level
```suggestion
```

File: keyboards/treadstone32/rev1/config.h [hunk 5]
```diff
@@ -84,20 +93,21 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+
+  #undef MOUSEKEY_MAX_SPEED
+  #define MOUSEKEY_MAX_SPEED 3
+
+  #undef MOUSEKEY_MOVE_DELTA
+  #define MOUSEKEY_MOVE_DELTA 4
+
+  #undef MOUSEKEY_DELAY
+  #define MOUSEKEY_DELAY 0
+#endif
```
Comment (zvecr) [inline keyboards/treadstone32/rev1/config.h:113]: Should really use defaults where possible at the keyboard level
```suggestion
```

File: keyboards/treadstone32/rev1/rev1.c (hunk 1 above)
Comment (zvecr) [reply keyboards/treadstone32/rev1/rev1.c:43]: Same change can also be applied to `lite.c`

Comment (marksard) [reply keyboards/treadstone32/rev1/rev1.c]: thanks, remove it.

File: keyboards/treadstone32/readme.md (hunk 2 above)
Comment (marksard) [reply keyboards/treadstone32/readme.md]: thanks

File: keyboards/treadstone32/rules.mk (hunk 3 above)
Comment (marksard) [reply keyboards/treadstone32/rules.mk:30]: thanks

File: keyboards/treadstone32/lite/config.h (hunk 4 above)
Comment (marksard) [reply keyboards/treadstone32/lite/config.h:112]: I used to set this to a per-keymap setting, so I'll undo it.

**Comment by marksard** [comment]
Any other changes?

File: keyboards/treadstone32/lite/config.h [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+//   #define RGBLIGHT_EFFECT_CHRISTMAS
+  #define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//   #define RGBLIGHT_EFFECT_RGB_TEST
+//   #define RGBLIGHT_EFFECT_ALTERNATING
+/*== customize breathing effect ==*/
+  /*==== (DEFAULT) use fixed table instead of exp() and sin() ====*/
+  #define RGBLIGHT_BREATHE_TABLE_SIZE 256      // 256(default) or 128 or 64
+  /*==== use exp() and sin() ====*/
+  #define RGBLIGHT_EFFECT_BREATHE_CENTER 1.85  // 1 to 2.7
+  #define RGBLIGHT_EFFECT_BREATHE_MAX    255   // 0 to 255
```
Comment (mtei) [inline keyboards/treadstone32/lite/config.h:80]: If you comment out these two lines, the firmware size may be reduced. The function remains the same, so try it out.

この2行をコメントアウトすると、ファームウェアサイズが小さくなるかもしれません。機能は変わらないので試してみてください。

Comment (marksard) [reply keyboards/treadstone32/lite/config.h:80]: ありがとうございます。
RGBLIGHT_ENABLE = yes状態でコメントアウト前後で比較しましたが、前後とも同じサイズでした。
```
 * The firmware size is fine - 19782/28672 (68%, 8890 bytes free)
```

Comment (mtei) [reply keyboards/treadstone32/lite/config.h:80]: RGBLIGHT_EFFECT_BREATHING を使っている時に有効な節約法でした。使ってなかったんですね。OK.

**Comment by marksard** [comment]
Could you tell me why this PR is never end?

**Comment by noroadsleft** [comment]
Thanks! Sorry about the delays; personally, I've been swamped with work for the Breaking Changes cycle, so I haven't reviewed anything outside of that for the last week or so.

**Pull request merged**